ddl/ddl_worker.go

```diff
@@ -614,6 +664,144 @@ func (w *JobContext) setDDLLabelForDiagnosis(job *model.Job) {
 	w.ddlJobCtx = kv.WithInternalSourceType(w.ddlJobCtx, w.ddlJobSourceType())
 }
 
+func (w *worker) HandleJobDone(d *ddlCtx, job *model.Job, t *meta.Meta) error {
+	err := w.finishDDLJob(t, job)
+	if err != nil {
+		w.sess.rollback()
+		return err
+	}
+
+	err = w.sess.commit()
+	if err != nil {
+		return err
+	}
+	asyncNotify(d.ddlJobDoneCh)
+	return nil
+}
+
+func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) error {
+	var (
+		err       error
+		schemaVer int64
+		runJobErr error
+		waitTime  = 2 * d.lease
+	)
+	defer func() {
+		w.unlockSeqNum(err)
+	}()
+
+	err = w.sess.begin()
+	if err != nil {
+		return err
+	}
+	if !variable.EnableConcurrentDDL.Load() || d.waiting.Load() {
```

Comment:
If `d.waiting.Load()` is true, we will exit `HandleDDLJobTable`. And then we'll go back to this logic and go to `getDDLJobAndRun` ->  `delivery2worker` -> `insertRunningDDLJobMap`? 
In this case, if the concurrency is large, is it likely that "runningJobs" will never be empty? But there is no worker to handle the job?

Reply:
no, it will be waiting before `getDDLJobAndRun`, see `job_table.goL171`

Reply:
So in non-concurrent mode, don't we need to do `insertRunningDDLJobMap`?

Reply:
yes, we should record the job id in non-concurrent mode